### PR TITLE
Centrify Credential Plugin for AWX/Tower

### PR DESCRIPTION
**Pull request by surbhijain1502**
##### SUMMARY
External Credential KV Plugin developed to support Centrify Vault, following the other credential plugins designs for AWX. 

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API
 - UI

##### AWX VERSION
awx: 15.0.1

### CODE REVIEW

File: awx/main/credential_plugins/centrify_vault.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        'type':'string',
+        'help_text': _('Password of Centrify API User with necessary permissions'),
+        'secret':True,
+    }],
+    'metadata': [{
+        'id': 'account-name',
+        'label': _('Account Name'),
+        'type': 'string',
+        'help_text': _('Local system account or Domain account name enrolled in Centrify Vault. eg. (root or DOMAIN/Administrator)'),
+        'secret': True,
```
Comment (srinivasgajavada) [inline awx/main/credential_plugins/centrify_vault.py:32]: Account name need not be secret (as discussed in the demo).

File: awx/main/credential_plugins/centrify_vault.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        return tokens['access_token']
+    except KeyError:
+        raise RuntimeError('OAuth request to tenant was unsuccessful')
+
+
+# fetch the ID of system with RedRock query, Input : System Name, Account Name
+def get_ID(**kwargs):
+    endpoint = urljoin(kwargs['url'],'/Redrock/query')
+    name=" Name='{0}' and User='{1}'".format(kwargs['system_name'],kwargs['acc_name'])
+    query  = 'Select * from VaultAccount where {0}'.format(name)
```
Comment (srinivasgajavada) [inline awx/main/credential_plugins/centrify_vault.py:72]: You can possibly reduce querying all the columns (*) here,  just reduce it to ID, Name ?

Comment (surbhijain1502) [reply awx/main/credential_plugins/centrify_vault.py]: Resolved, Just ID required

File: awx/main/credential_plugins/centrify_vault.py (hunk 1 above)
Comment (surbhijain1502) [reply awx/main/credential_plugins/centrify_vault.py]: Done.

**Review by badrogh — APPROVED** [review]

**Review by badrogh — APPROVED** [review]

**Review by albert-leung — APPROVED** [review]

**Review by pdakoju2020 — APPROVED** [review]

**Pull request merged**